packages/teams-js/src/public/dialog.ts

```diff
@@ -111,183 +88,383 @@ export namespace dialog {
    *
-   * @returns boolean to represent whether dialog module is supported
+   * @internal
    */
-  export function isSupported(): boolean {
-    return runtime.supports.dialog ? true : false;
+  export function getDialogInfoFromBotAdaptiveCardDialogInfo(
+    botAdaptiveCardDialogInfo: BotAdaptiveCardDialogInfo,
+  ): DialogInfo {
+    const dialogInfo: DialogInfo = {
+      card: botAdaptiveCardDialogInfo.card,
+      height: botAdaptiveCardDialogInfo.size ? botAdaptiveCardDialogInfo.size.height : DialogDimension.Small,
+      width: botAdaptiveCardDialogInfo.size ? botAdaptiveCardDialogInfo.size.width : DialogDimension.Small,
+      title: botAdaptiveCardDialogInfo.title,
+      completionBotId: botAdaptiveCardDialogInfo.completionBotId,
+    };
+    return dialogInfo;
   }
 
   /**
-   * Namespace to update the dialog
+   * Subcapability for interacting with adaptive card dialogs
    */
-  export namespace update {
+  export namespace adaptiveCard {
     /**
-     * Update dimensions - height/width of a dialog.
+     * Allows app to open an adaptive card based dialog.
      *
-     * @param dimensions - An object containing width and height properties.
+     * @remarks
+     * This function cannot be called from inside of a dialog
+     *
+     * @param adaptiveCardDialogInfo - An object containing the parameters of the dialog module.
+     * @param submitHandler - Handler that triggers when a dialog calls the {@linkcode submit} function or when the user closes the dialog.
      */
-    export function resize(dimensions: DialogSize): void {
-      ensureInitialized(FrameContexts.content, FrameContexts.sidePanel, FrameContexts.task, FrameContexts.meetingStage);
+    export function open(adaptiveCardDialogInfo: AdaptiveCardDialogInfo, submitHandler?: DialogSubmitHandler): void {
+      ensureInitialized(FrameContexts.content, FrameContexts.sidePanel, FrameContexts.meetingStage);
       if (!isSupported()) {
         throw errorNotSupportedOnPlatform;
       }
-      sendMessageToParent('tasks.updateTask', [dimensions]);
+
+      getVersion().then(hostVersion => {
+        validateMinimumAdaptiveCardVersion(hostVersion);
+
+        const dialogInfo: DialogInfo = getDialogInfoFromAdaptiveCardDialogInfo(adaptiveCardDialogInfo);
+        sendMessageToParent('tasks.startTask', [dialogInfo], (err: string, result: string | object) => {
+          submitHandler?.({ err, result });
+        });
+      });
     }
 
     /**
-     * Checks if dialog.update capability is supported by the host
+     * Query the host to see what version of the AdaptiveCard schema they support
      *
-     * @returns boolean to represent whether dialog.update is supported
+     * @returns a Promise containing the {@linkcode AdaptiveCardVersion}, describing what version of the AdaptiveCard
+     * schema the host supports
+     */
+    export function getVersion(): Promise<AdaptiveCardVersion> {
```

Comment:
I believe this should be in a top-level `adaptiveCards` capability, unless there are scenarios where different capabilities leveraging ACs will somehow be backed by a different version.

Reply:
Removing and replacing with adding to intialization flow